packages/containers/src/List/List.container.js

```diff
@@ -178,36 +180,37 @@ class List extends React.Component {
 			const actions = this.props.actions;
 			if (actions) {
 				if (actions.left) {
-					props.toolbar.actionBar.actions.left = actions.left.map(action => ({ name: action }));
+					props.toolbar.actionBar.actions.left = actions.left.map(action => ({ actionId: action }));
 				}
 				if (actions.right) {
-					props.toolbar.actionBar.actions.right = actions.right.map(action => ({ name: action }));
+					props.toolbar.actionBar.actions.right = actions.right.map(action => ({
+						actionId: action,
+					}));
 				}
 			}
 
-			const pagination = this.props.toolbar.pagination;
-			if (pagination) {
-				props.toolbar.pagination = {
-					...pagination,
-					totalResults: state.totalResults,
-					itemsPerPage: state.itemsPerPage,
-					startIndex: state.startIndex,
-					onChange: (startIndex, itemsPerPage) => {
-						if (pagination.onChange) {
-							this.props.dispatchActionCreator(
-								pagination.onChange,
-								null,
-								{ startIndex, itemsPerPage },
-								this.context,
-							);
-						} else {
-							this.onChangePage(startIndex, itemsPerPage);
-						}
-					},
-				};
+			if (props.toolbar.pagination) {
+				const pagination = props.toolbar.pagination;
```

Comment:
could replace this by:

pagination = {
... props.toolbar.pagination,
...pick(state, ['totalResults', 'itemsPerPage', 'startIndex']
}

more reader no ?

Reply:
yes but it break the reference so it will not update props.toolbar.pagination